File: TypeScript/27.ts
function removeElement(nums: number[], val: number): number {
  let k = 0;
  nums.forEach((item) => {
    if (item !== val) {
      let temp = item;
      item = nums[k];
      nums[k] = temp;
      ++k;
    }
  });
  return k;
}

console.log(removeElement([3, 2, 2, 3], 3)); // TEST: 2

File: C++/src/27.cpp
#include <solution_tmpl.h>

int Solution::removeElement(vector<int> &nums, int val) {
  int k = 0;
  for (size_t i = 0; i < nums.size(); i++) {
    if (nums[i] != val) {
      nums[k] = nums[i];
      k++;
    }
  }
  return k;
}
